src/csharp/Microsoft.Spark/Sql/RowCollector.cs

```diff
@@ -33,7 +34,29 @@ public IEnumerable<Row> Collect(ISocketWrapper socket)
 
                 foreach (object unpickled in unpickledObjects)
                 {
-                    yield return (unpickled as RowConstructor).GetRow();
+                    // yield return (unpickled as RowConstructor).GetRow();
+                    switch (unpickled)
```

Comment:
Can you add comments what's happening?

Reply:
This actually returns object array of row. I was wondering if this is caused by casting generic row to object?

Reply:
Can we understand what's really happening before moving forward? (and add comments to the code)

Reply:
I have added one line comments. Let me know if this makes to you. Thanks!